doc/website/blog/2023-02-23-simple-cms-with-ent.mdx

```diff
@@ -0,0 +1,779 @@
+}
+
+```
+
+Next, let's define the handler for our blog home page. This page should contain a list of all available blog posts:
+
+```go
+// index serves the blog home page
+func (s *server) index(w http.ResponseWriter, r *http.Request) {
+	posts, err := s.client.Post.
+		Query().
+		WithAuthor().
+		All(r.Context())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if err := tmpl.Execute(w, posts); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+```
+
+Let's zoom in on the Ent code here that is used to retrieve the posts from the database:
+```go
+// s.client.Post contains methods for interacting with Post entities
+s.client.Post.
+	// Begin a query.
+	Query().
+	// Retrieve the entities using the `Author` edge. (a `User` instance)
+	WithAuthor().
+	// Run the query against the database using the request context.
+	All(r.Context())
+```
+
+#### The router
+
+To manage the routes for our application, let's use `go-chi`, a popular routing library for Go.
+
+```
+go get -u github.com/go-chi/chi/v5
+```
+
+We define the `newRouter` function that sets up our router:
+
+```go
+// newRouter creates a new router with the blog handlers mounted.
+func newRouter(srv *server) chi.Router {
+	r := chi.NewRouter()
+	r.Use(middleware.Logger)
+	r.Use(middleware.Recoverer)
+	r.Get("/", srv.index)
+	return r
+}
+```
+
+In this function, we first instantiate a new `chi.Router`, then register two middlewares:
+* `middleware.Logger` is a basic access logger that prints out some information on every request our server handles.
+* `middleware.Recoverer` recovers from when our handlers panic, preventing a case where our entire server will crash because of an application error.
+
+Finally, we register the `index` function of the `server` struct to handle `GET` requests to the `/` path of our server.
+
+#### A unit test
+
+Before wiring everything together, let's write a simple unit test to check that our code works as expected.
+
+To simplify our tests we will install the SQLite driver for Go which allows us to use an in-memory database:
+```
+go get -u github.com/mattn/go-sqlite3
+```
+
+Next, we install `testify`, a utility library that is commonly used for writing assertions in tests.
+
+```
+go get github.com/stretchr/testify 
+```
+
+With these dependencies installed, create a new file named `main_test.go`:
+
+```go
+package main
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/rotemtam/ent-blog-example/ent/enttest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIndex(t *testing.T) {
+	// Initialize an Ent client that uses an in memory SQLite db.
+	client := enttest.Open(t, "sqlite3", "file:ent?mode=memory&cache=shared&_fk=1")
+	defer client.Close()
+	
+	// seed the database with our "Hello, world" post and user.
+	err := seed(context.Background(), client)
+	require.NoError(t, err)
```

Comment:
```suggestion
	require.NoError(t, seed(context.Background(), client))
```

Reply:
I actually prefer it this way. I don't think there's a benefit from condensing it to one line, if anything, its harder to read.